sbncode/EventGenerator/MeVPrtl/Tools/HNL/AnisotropicThreeBodyDecay.cpp

```diff
@@ -0,0 +1,510 @@
+#include "AnisotropicThreeBodyDecay.h"
+#include "sbncode/EventGenerator/MeVPrtl/Tools/Constants.h"
+
+#include "TLorentzVector.h"
+#include "TRandom3.h"
+
+//Implementation of HNL Three Body Decays Anisotropies
+//Valid as long as the HNL is decaying into a neutrino and identical final-state charged leptons
+//See arXiv:2104.05719 for more details @LuisPelegrina
+
+
+//Constants used for the calculations of C_i Coefficients given the electro-weak mixing angle
+//See arXiv:2104.05719 (Table 3) for more details
+double evgen::ldm::AnThreeBD::GL(bool CC)
+{
+  double const sw2=Constants::Instance().sin2thetaW;
+  if(CC)  return 0.5*(1+2*sw2); //When the decay can happen by CC gL changes
+  else  return -0.5*(1-2*sw2);
+}
+
+double evgen::ldm::AnThreeBD::GR()
+{
+  double const sw2=Constants::Instance().sin2thetaW;
+  return sw2;
+}
+
+
+//Coefficients of Lorentz-Invariant Objects given couplings gL and gR (Majorana HNL)
+//See arXiv:2104.05719 (Table 3) for more details
+//Valid as long as the HNL is decaying into a neutrino and identical final-state charged leptons 
+void evgen::ldm::AnThreeBD::CMgLgR(double (&CM)[6],double Ue4,double Umu4,double Ut4, int LeptonPDG)
+{
+  double U4[3]={Ue4,Umu4,Ut4};
+  
+  for (int i=0;i<6;i++)
+    {
+      CM[i]=0;
+    }
+
+  double gl;
+  double gr=GR();
+  //Calculate the contribution of each mixing matrix element and sum them 
+  for (int i=0;i<3;i++)
+    {
+      if (LeptonPDG==11+2*i) gl= GL(true);// Take into account the CC contribution 
+      else gl=GL(false); 
+      CM[0]+=128*U4[i]*gr*gl; //C1
+      CM[1]+=64*U4[i]*(gl*gl+gr*gr);//C4
+      CM[2]+=64*U4[i]*(gl*gl+gr*gr);//C5
+      CM[3]+=0;//C8
+      CM[4]+=64*U4[i]*(gr*gr-gl*gl);//C9
+      CM[5]+=-64*U4[i]*(gr*gr-gl*gl);//C10
+    }
+    
+  return;
+}
+
+//Coefficients of Lorentz-Invariant Objects given couplings gL and gR (Dirac HNL)
+//See arXiv:2104.05719 (Table 3) for more details
+//Valid as long as the HNL is decaying into a neutrino and identical final-state charged leptons 
+void evgen::ldm::AnThreeBD::CDNgLgR(double (&CD)[6],double Ue4,double Umu4,double Ut4, int LeptonPDG)
+{
+  double U4[3]={Ue4,Umu4,Ut4};
+  
+  for (int i=0;i<6;i++)
+    {
+      CD[i]=0;
+    }
+
+  double gl;
+  double gr=GR();
+  //Calculate the contribution of each mixing matrix element and sum them 
+  for (int i=0;i<3;i++)
```

Comment:
Table 3 from the reference only applies to the case of nu_mu - N mixing. This for loop over all matrix elements is not correct.